requirements.txt

```diff
@@ -1,8 +1,7 @@
 black==22.3.0
-duneapi==3.0.3
+duneapi==3.0.6
 mypy==0.942
 pylint==2.13.4
 pytest==7.1.1
 python-dotenv==0.20.0
-types-requests==2.27.16
-web3>=5.28.0
```

Comment:
Notice that this direct dependency was removed since it was only used for the address class, this is now a dependency of duneapi package.

Reply:
However, it appears our CI fails because it requires web3 here. Is this the way things are supposed to work? Why should I need to explicitly declare sub dependencies?

Reply:
I will put this dependency back for now, but I find it rather strange how this works...

Reply:
It turned out I need to explicitly specify `install_requires` in the `setup.py` file! Really nice, now we can remove a few requirements here since they are indirect.